app/assets/javascripts/service_uploads/service_uploads_page.js

```diff
@@ -106,47 +105,41 @@
       this.setState({
         formData: merge(this.state.formData, { date_started: event })
       });
-
-      this.verifyRequiredFields();
     },
 
     onSelectEndDate: function (event) {
       this.setState({
         formData: merge(this.state.formData, { date_ended: event })
       });
-
-      this.verifyRequiredFields();
     },
 
     onUserTypingServiceType: function(event) {
       this.setState({
         formData: merge(this.state.formData, { service_type_name: event.target.value })
       });
-
-      this.verifyRequiredFields();
     },
 
     onUserSelectServiceType: function(string) {
       this.setState({
         formData: merge(this.state.formData, { service_type_name: string })
       });
-
-      this.verifyRequiredFields();
     },
 
     onClickUploadButton: function () {
-      if (this.state.missingRequiredFields === true) return;
+      if (this.isMissingRequiredFields()) return;
 
       this.upload();
     },
 
-    verifyRequiredFields: function () {
+    isMissingRequiredFields: function () {
```

Comment:
@alexsoble Switching this to a method that computes the value for "is anything missing?" means that the various other methods here don't all have to make sure to update `missingFields`.